ops_openstack.py

```diff
@@ -87,7 +87,9 @@ def update_status(self):
             return
         missing_relations = []
         for relation in self.REQUIRED_RELATIONS:
-            if not self.model.get_relation(relation):
+            try:
+                self.model.get_relation(relation)
+            except KeyError:
```

Comment:
get_relation will only raise `KeyError` if the relation name is not in `metadata.yaml`.
https://github.com/canonical/operator/blob/c64b2f51576cb6bee9a8a2d9fd566592184d7abb/ops/model.py#L58

* If there is no relation yet, it will return `None`;
* if there are multiple relations, it will raise `TooManyRelatedAppsError` (because it's a helper function to avoid doing self.model.relations[name][0] for apps that don't support multiple related remote apps).

Reply:
ok, I changed this due to the results of unit testing but it turns out there was a bug in the operator framework test harness. I've raised an issue ( https://github.com/canonical/operator/issues/241 ) and reverted back to the original behaviour